### PR TITLE
Remove any `wp_reset_vars()` usage in core

### PR DESCRIPTION
**Pull request by swissspidy**
Trac ticket: https://core.trac.wordpress.org/ticket/38073

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-wp-links-list-table.php [hunk 1]
```diff
@@ -50,7 +50,10 @@ public function ajax_user_can() {
 	public function prepare_items() {
 		global $cat_id, $s, $orderby, $order;
 
-		wp_reset_vars( array( 'action', 'cat_id', 'link_id', 'orderby', 'order', 's' ) );
```
Comment (swissspidy) [inline src/wp-admin/includes/class-wp-links-list-table.php:53]: Turns out `action` and `link_id` aren't really used here.

File: src/wp-admin/includes/deprecated.php [hunk 2]
```diff
@@ -1589,3 +1589,31 @@ function image_attachment_fields_to_save( $post, $attachment ) {
...
+ * This function resets global variables based on the names passed
+ * in the $vars array to the value of $_POST[$var] or $_GET[$var] or ''
+ * if neither is defined.
+ *
+ * @since 2.0.0
+ * @deprecated 6.6.0
+ *
+ * @param array $vars An array of globals to reset.
+ */
+function wp_reset_vars( $vars ) {
```
Comment (swissspidy) [inline src/wp-admin/includes/deprecated.php]: According to https://wpdirectory.net/search/01HWQEZAZ2VMB65S0FBX80HNZB there are some plugins using the function in their custom list tables. So maybe we shouldn't deprecate the function. We should still remove its usage in core though.

**Review by audrasjb — APPROVED** [review]
I approve the change as technically, the patch looks good to me. But we have to decide whether the function is deprecated or not. I left a comment in the ticket thread.

**Comment by swissspidy** [comment]
Committed in https://core.trac.wordpress.org/changeset/58069

**Pull request closed without merging**